rpc/lighthouse.go

```diff
@@ -52,55 +52,65 @@ func NewLighthouseClient(endpoint string, chainID *big.Int) (*LighthouseClient,
 }
 
 func (lc *LighthouseClient) GetNewBlockChan() chan *types.Block {
-	// setup health check & exit if the new block chan does no longer receive new blocks
+	// setup health check & reconnect if the new block chan no longer receive new blocks
 	lc.lastBlockSeenMux.Lock()
 	lc.lastBlockSeen = time.Now()
 	lc.lastBlockSeenMux.Unlock()
+	reConnect := make(chan int64, 10)
+	reConnectValue := int64(0)
 	go func() {
 		for ; ; time.Sleep(time.Second) {
 			lc.lastBlockSeenMux.Lock()
-			if time.Since(lc.lastBlockSeen) > time.Minute*2 {
-				lc.lastBlockSeenMux.Unlock()
-				// fatal if no new block was received for more than two minutes
-				logger.Fatalf("lighthouse client error, no new block retrieved since %v (%v ago)", lc.lastBlockSeen, time.Since(lc.lastBlockSeen))
+			if reConnectValue == 0 {
+				reConnectValue++
+				reConnect <- reConnectValue
+			} else if time.Since(lc.lastBlockSeen) > time.Minute*2 {
+				// if we did not receive a new block more than two minutes we try to reconnect to the lighthouse node
+				logger.Errorf("lighthouse client error, no new block retrieved since %v (%v ago)", lc.lastBlockSeen, time.Since(lc.lastBlockSeen))
+				lc.lastBlockSeen = time.Now()
```

Comment:
**Suggestion:** Personally, I would not touch `lastBlockSeen` here. I would send the `reConnect` command and wait for 10s (or so) after that. Setting `lastBockSeen` here even though no block was seen feels like a misuse.

Please share your thoughts.

Reply:
removed